include/VectorGraph.h

```diff
@@ -0,0 +1,739 @@
+	void setDataArray(std::vector<float>* dataArrayIn, bool shouldCurve, bool shouldClear, bool shouldClamp, bool shouldRescale, bool callDataChanged);
+	void setDataArray(float* dataArrayIn, unsigned int sizeIn, bool shouldCurve, bool shouldClear, bool shouldClamp, bool shouldRescale, bool callDataChanged);
+
+
+	// set attribute: -------------------
+	// sets position when m_isFixedX is disabled, returns final location
+	unsigned int setX(unsigned int pointLocation, float newX);
+	// sets value when m_isFixedY is disabled
+	void setY(unsigned int pointLocation, float newY);
+	// sets value when m_isEditableAttrib is enabled
+	void setC(unsigned int pointLocation, float newC);
+	// sets value when m_isEditableAttrib is enabled
+	void setValA(unsigned int pointLocation, float fValue);
+	// sets value when m_isEditableAttrib is enabled
+	void setValB(unsigned int pointLocation, float fValue);
+	// sets value when m_isEditableAttrib is enabled
+	void setType(unsigned int pointLocation, unsigned int newType);
+	// sets what attribute gets automated when m_isEditableAttrib and m_isAutomatableEffectable is enabled
+	void setAutomatedAttrib(unsigned int pointLocation, unsigned int attribLocation);
+	// sets what attribute gets effected when m_isEditableAttrib and m_isAutomatableEffectable is enabled
+	void setEffectedAttrib(unsigned int pointLocation, unsigned int attribLocation);
+	void setEffectOnlyPoints(unsigned int pointLocation, bool bValue);
+	void setEffect(unsigned int pointLocation, unsigned int effectId, bool bValue);
+	// if bValue is true then make a new FloatModel and connect it, else delete
+	// the currently used FloatModel
+	// runs if m_isAutomatableEffectable is enabled
+	void setAutomated(unsigned int pointLocation, bool bValue);
+
+
+// signals: // not qt
+	void dataChanged();
+	// runs when m_dataArray.size() gets to 0
+	void clearedEvent();
+	// color
+	void styleChanged();
+protected:
+	// returns m_automationModelArray
+	std::vector<FloatModel*>* getAutomationModelArray();
+	// delete automationModels in m_automationModelArray
+	// that are not used by points (there should be 0 cases like this)
+	void delUnusedAutomation();
+	QString getSavedDataArray();
+	void loadDataArray(QString data, unsigned int arraySize);
+private:
+	class VectorGraphPoint
+	{
+	public:
+		inline VectorGraphPoint()
+		{
+			m_x = 0.0f;
+			m_y = 0.0f;
+			m_c = 0.0f;
+			m_valA = 0.0f;
+			m_valB = 0.0f;
+			m_type = 0;
+			m_automatedEffectedAttribLocations = 0;
+
+			m_effectOnlyPoints = false;
+			m_effectAdd = false;
+			m_effectSubtract = false;
+			m_effectMultiply = false;
+			m_effectDivide = false;
+			m_effectPower = false;
+			m_effectLog = false;
+			m_effectSine = false;
+			m_effectClampLower = false;
+			m_effectClampUpper = false;
+
+			m_bufferedAutomationValue = 0.0f;
+			m_automationModel = -1;
+		}
+		inline VectorGraphPoint(float x, float y)
+		{
+			m_x = x;
+			m_y = y;
```

Comment:
After moving the defaults out:
```suggestion
		inline VectorGraphPoint(float x, float y) : m_x(x), m_y(y) {}
```